### ISSUE TITLE
Flame solver does not retain transport options after automatically expanding domain

### ISSUE THREAD

**Issue by speth** [Cantera/cantera#615]
When solving the freely propagating flame with the `auto` option, if the solver has to expand the domain to fit the flame, the settings for `transport_model` and `soret_enabled` are not retained, but are replaced with the values used in the first solving stage of `'Mix'` and `False`.

Example:
```py
import cantera as ct
print("Running Cantera version", ct.__version__)

gas = ct.Solution('gri30.cti', 'gri30_multi')
gas.set_equivalence_ratio(1.0, 'CH4', {'O2':1.0, 'N2':3.76})
gas.TP = 300, 101325
width = 0.01  # domain that is too narrow

flame = ct.FreeFlame(gas, width=width)
flame.set_refine_criteria(ratio=3, slope=0.1, curve=0.1)

flame.transport_model = 'Multi'
flame.soret_enabled = True

flame.solve(loglevel=1, auto=True)

Su0 = flame.u[0]
print("Flame Speed is: {:.2f} cm/s".format(Su0*100))
flame.show_stats()
print(flame.transport_model)
print(flame.soret_enabled)
```
Changing the initial domain width to 0.06 m eliminates the need to expand the domain, and results in the use of the multicomponent transport model with the Soret effect enabled for the final solve. This affects version 2.4.0 and newer, since the automatic grid expansion was added in that version.

**Comment by ischoegl** [comment]
Behaviour is caused by the `auto` solution mechanism implemented in `Sim1D.solve`. Both soret and multicomponent are temporarily disabled; if an Exception occurs, they are not restored.

**Comment by ischoegl** [comment]
@speth ... feel free to review the proposed fix (pull request #631)
